src/main/java/spoon/support/compiler/jdt/JDTImportBuilder.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (C) 2006-2017 INRIA and contributors
+ * Spoon - http://spoon.gforge.inria.fr/
+ *
+ * This software is governed by the CeCILL-C License under French law and
+ * abiding by the rules of distribution of free software. You can use, modify
+ * and/or redistribute the software under the terms of the CeCILL-C license as
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.support.compiler.jdt;
+
+import org.eclipse.jdt.core.compiler.CharOperation;
+import org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration;
+import org.eclipse.jdt.internal.compiler.ast.ImportReference;
+import org.eclipse.jdt.internal.compiler.env.ICompilationUnit;
+import spoon.reflect.cu.CompilationUnit;
+import spoon.reflect.declaration.CtMethod;
+import spoon.reflect.declaration.CtNamedElement;
+import spoon.reflect.declaration.CtPackage;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.declaration.ModifierKind;
+import spoon.reflect.factory.Factory;
+import spoon.reflect.reference.CtFieldReference;
+import spoon.reflect.reference.CtReference;
+import spoon.reflect.visitor.filter.NameFilter;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Created by urli on 08/08/2017.
+ */
+public class JDTImportBuilder {
+
+	private final CompilationUnitDeclaration declarationUnit;
+	private String filePath;
+	private CompilationUnit spoonUnit;
+	private ICompilationUnit sourceUnit;
+	private Factory factory;
+	private Set<CtReference> imports;
+
+	JDTImportBuilder(CompilationUnitDeclaration declarationUnit,  Factory factory) {
+		this.declarationUnit = declarationUnit;
+		this.factory = factory;
+		this.sourceUnit = declarationUnit.compilationResult.compilationUnit;
+		this.filePath = CharOperation.charToString(sourceUnit.getFileName());
+		this.spoonUnit = factory.CompilationUnit().create(filePath);
+		this.imports = new HashSet<>();
+	}
+
+	public void build() {
+		if (declarationUnit.imports == null || declarationUnit.imports.length == 0) {
+			return;
+		}
+
+		for (ImportReference importRef : declarationUnit.imports) {
+			String importName = importRef.toString();
+			if (!importRef.isStatic()) {
+				// Starred import are only managed by importing types of the model for now
+				// as it can cost a lot to retrieve all classes of a package by reflection
+				if (importName.endsWith("*")) {
+					int lastDot = importName.lastIndexOf(".");
+					String packageName = importName.substring(0, lastDot);
+
+					CtPackage ctPackage = this.factory.Package().get(packageName);
```

Comment:
check that this does not use reflection too with shadow packages. if not document